projects/gloo/pkg/plugins/consul/subsets.go

```diff
@@ -42,6 +42,13 @@ func (p *plugin) ProcessRouteAction(params plugins.RouteActionParams, inAction *
 			Destinations: upstreamGroup.Destinations,
 		}
 		return setWeightedClusters(params.Params, md, out)
+
+	case *v1.RouteAction_ClusterHeader:
+		// ClusterHeader must use the naming convention {{namespace}}_{{clustername}}
+		out.ClusterSpecifier = &envoyroute.RouteAction_ClusterHeader{
+			ClusterHeader: dest.ClusterHeader,
```

Comment:
```suggestion
			ClusterHeader: dest.GetClusterHeader(),
```

In general whenever accessing data from a proto msg always use the getters as they are nil safe